core/network/protocols/gossip_protocol.cpp

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "network/protocols/gossip_protocol.hpp"
+
+#include <libp2p/connection/loopback_stream.hpp>
+
+#include "network/common.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::network, GossipProtocol::Error, e) {
+  using E = kagome::network::GossipProtocol::Error;
+  switch (e) {
+    case E::GONE:
+      return "Protocol was switched off";
+  }
+  return "Unknown error";
+}
+
+namespace {
+  using libp2p::connection::LoopbackStream;
```

Comment:
Are you sure 'using' influences linkage?

Reply:
Of course not